### PR TITLE
feature(authoption): generic Auth Options

### PR DESCRIPTION
**Pull request by udaydenduluri33**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/auth/models/optionsMapper.class.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            scope: openIdOptions.scope,
+            sessionKey: openIdOptions.sessionKey,
+            token_endpoint_auth_method: openIdOptions.tokenEndpointAuthMethod as ClientAuthMethod,
+            useRoutes: false,
+        }
+        /* eslint-disable @typescript-eslint/camelcase */
+        return options
+    }
+
+    public static getOAuth2Options(oAuth2Options: AuthOptions): OAuth2Metadata {
```
Comment (paddybasi) [inline src/auth/models/optionsMapper.class.ts:26]: I think, we don't actually need this one, the options going in are the same as the ones coming out

Comment (udaydenduluri33) [reply src/auth/models/optionsMapper.class.ts:26]: Agreed. Remove it

File: src/auth/models/authOptions.interface.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+export interface AuthOptions {
+    authorizationURL: string
```
Comment (Daniel-Lam) [inline src/auth/models/authOptions.interface.ts:2]: Capitalisation consistency... in some places, the names use `somethingURL` and in others it's `somethingUrl` - should it not be one or the other?

File: src/auth/models/optionsMapper.class.spec.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    expect(openIdOptions.discovery_endpoint).toEqual(options.discoveryEndpoint)
+    expect(openIdOptions.issuer_url).toEqual(options.issuerUrl)
+    expect(openIdOptions.logout_url).toEqual(options.logoutUrl)
+    expect(openIdOptions.redirect_uri).toEqual(options.callbackURL)
+    expect(openIdOptions.response_types).toEqual(options.responseTypes)
+    expect(openIdOptions.scope).toEqual(options.scope)
+    expect(openIdOptions.sessionKey).toEqual(options.sessionKey)
+    expect(openIdOptions.token_endpoint_auth_method).toEqual(options.tokenEndpointAuthMethod)
+    expect(openIdOptions.useRoutes).toEqual(options.useRoutes)
+})
```
Comment (Daniel-Lam) [inline src/auth/models/optionsMapper.class.spec.ts:32]: How come some values aren't being checked? For example, what about the mapping for `options.authorizationURL` and `options.tokenURL`?

Comment (udaydenduluri33) [reply src/auth/models/optionsMapper.class.spec.ts:32]: openIdOptions do not have these options. authorizationURL and tokenURL. It uses discovery endpoints.

File: src/auth/models/authOptions.interface.ts (hunk 2 above)
Comment (udaydenduluri33) [reply src/auth/models/authOptions.interface.ts:2]: Done

File: src/auth/models/optionsMapper.class.spec.ts (hunk 3 above)
Comment (Daniel-Lam) [reply src/auth/models/optionsMapper.class.spec.ts:32]: Ah, OK - that's fine

File: src/auth/models/authOptions.interface.ts (hunk 2 above)
Comment (Daniel-Lam) [reply src/auth/models/authOptions.interface.ts]: Resolved - @udaydenduluri33 explained that `tokenURL` and `callbackURL` are inherited from a base class in a library outside of our control.

**Pull request merged**